include/range/v3/view/bounded.hpp

```diff
@@ -41,65 +41,82 @@ namespace ranges
           : view_interface<bounded_view<Rng>, range_cardinality<Rng>::value>
         {
         private:
+            CONCEPT_ASSERT(Range<Rng>());
+            CONCEPT_ASSERT(!BoundedRange<Rng>());
+
+            template<typename R>
             using RA_and_Sized =
-                meta::strict_and<RandomAccessRange<Rng>, SizedRange<Rng>>;
-            using base_iterator_t = range_iterator_t<Rng>;
-            using base_sentinel_t = range_sentinel_t<Rng>;
+                meta::strict_and<RandomAccessRange<R>, SizedRange<R>>;
+            template<typename R>
+            using bounded_iterator_t = meta::if_<RA_and_Sized<R>,
+                range_iterator_t<R>,
+                common_iterator<range_iterator_t<R>, range_sentinel_t<R>>>;
+            template<typename R>
+            using bounded_iterator = meta::defer<bounded_iterator_t, R>;
```

Comment:
...and move these non-dependent aliases out into the detail namespace while you're at it.